Kiosk/ServiceHelpers/InkRecognizer.cs

```diff
@@ -0,0 +1,152 @@
+﻿// 
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// 
+// Microsoft Cognitive Services: http://www.microsoft.com/cognitive
+// 
+// Microsoft Cognitive Services Github:
+// https://github.com/Microsoft/Cognitive
+// 
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+// 
+// MIT License:
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED ""AS IS"", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Data.Json;
+using Windows.Graphics.Display;
+using Windows.UI.Input.Inking;
+
+namespace ServiceHelpers
+{
+    public class InkRecognizer
```

Comment:
Couple of things:

- Use ServiceBase and HttpClientUtility classes. Among other things, this will make sure the sleep-retry pattern is used when the api keys reach quota limit exceeded errors.
- Initialize the endpoint variable in this class directly. Since Ink Recognizer is a global service, we don't need to offer the user an option for selecting the end point in the settings page (see my other comment).

Please see Kiosk/ServiceHelpers/TranslatorTextService.cs for a similar example with another service.

Reply:
Switched over to utilizing ServiceBase and HttpClientUtility classes as well as only using the global endpoint for the Ink Recognizer.